### PR TITLE
[DOCS] Clarify upgrade paths

### PR DESCRIPTION
**Pull request by lucabelluccini**
Unsupported upgrade paths:
- `6.8 to 7.0`.

Supported, but requires a full cluster restart:
- `6.0–6.7 directly to 7.x`

As discussed with @DaveCTurner and @tvernum , I've submitted those changes which should be reviewed.

This should be probably backported.

### CODE REVIEW

File: docs/reference/upgrade.asciidoc
Comment (jrodewig) [inline docs/reference/upgrade.asciidoc]:
```suggestion
In-place downgrades to earlier versions are *not* supported. To downgrade to an
```

File: docs/reference/upgrade.asciidoc
Comment (jrodewig) [inline docs/reference/upgrade.asciidoc]:
```suggestion
The following upgrade path is supported but requires a <<restart-upgrade,full cluster restart upgrade>>.
```

File: docs/reference/upgrade.asciidoc
Comment (jrodewig) [inline docs/reference/upgrade.asciidoc]:
```suggestion
The following upgrade path is *not* supported:
```

File: docs/reference/upgrade.asciidoc
Comment (jrodewig) [inline docs/reference/upgrade.asciidoc]: I don't think the emphasis is needed here.
```suggestion
The following table shows the recommended upgrade paths to {version}.
```

**Review by jrodewig — APPROVED** [review]
Thanks for making these clarifications @lucabelluccini.

I've made a few minor suggestions, but I don't consider any of them to be blocking.

File: docs/reference/upgrade.asciidoc
Comment (DaveCTurner) [inline docs/reference/upgrade.asciidoc]: I don't think a one-item list makes sense here. Can you fold it into a single sentence?

File: docs/reference/upgrade.asciidoc
Comment (DaveCTurner) [inline docs/reference/upgrade.asciidoc]: I don't think this bit should be part of this `WARNING`. I think it's adequately covered by other text on this page.

File: docs/reference/upgrade.asciidoc
Comment (DaveCTurner) [inline docs/reference/upgrade.asciidoc]: Suggest making this a separate `WARNING` -- it's unrelated to the first bit.

File: docs/reference/upgrade.asciidoc
Comment (DaveCTurner) [inline docs/reference/upgrade.asciidoc]: I think we should leave this here. It's more general than the text in the `WARNING` block above (6.7 _or earlier_) and I don't think it needs to be part of a `WARNING` at all.

**Review by DaveCTurner — CHANGES_REQUESTED** [review]
I left some comments.

File: docs/reference/upgrade.asciidoc
Comment (lucabelluccini) [reply docs/reference/upgrade.asciidoc]: I would keep it because it makes more explicit we're not talking only about rolling upgrades (as most of the page is).

File: docs/reference/upgrade.asciidoc
Comment (lucabelluccini) [reply docs/reference/upgrade.asciidoc]: If it is ok for you, I've kept it but moved up closed to the other upgrade paths.

File: docs/reference/upgrade.asciidoc
Comment (lucabelluccini) [reply docs/reference/upgrade.asciidoc]: I've added a separate warning.

**Comment by tvernum** [comment]
I like the most recent version of this change (70b103b), thanks.

I'll mark an approval in a moment, but please wait for @DaveCTurner's review, because he and I are coming at this from different angles.

The reason for my concern about having explicit information about 6.7 to {version} is that the 7.0 docs says that a rolling upgrade from 6.7 to 7.0.1 is OK. And it is.
I know from experience that some users start their 7.x upgrade planning by reading the release notes/instructions for 7.0 because in their mind the planning process is "what are the steps to go from 6 to 7, and then what specific steps are needed for 7.6"
If they read the 7.0 docs first then they have in their mind that 6.7 to 7.something is a rolling upgrade, and we need to be clear that it's only true for 7.0 and not for 7.6

I believe the current proposal does that.

**Review by tvernum — APPROVED** [review]

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Comment by lucabelluccini** [comment]
Hello all! Thank you for the review. I think I can squash & merge? How we will manage the backport?

**Comment by DaveCTurner** [comment]
Good point @lucabelluccini, I'll take care of it. Since this is against the 7.6 branch it'll need forward-porting after merging.

**Pull request merged**
